appservice/src/editScmType.ts

```diff
@@ -4,23 +4,28 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { WebSiteManagementModels } from 'azure-arm-website';
-import { AzureTreeItem, IAzureQuickPickItem, IAzureQuickPickOptions, UserCancelledError } from 'vscode-azureextensionui';
+import * as git from 'simple-git/promise';
+import { RemoteWithRefs } from 'simple-git/typings/response';
+import { AzureParentTreeItem, IAzureQuickPickItem, IAzureQuickPickOptions, UserCancelledError } from 'vscode-azureextensionui';
 import { connectToGitHub } from './connectToGitHub';
 import { ext } from './extensionVariables';
 import { localize } from './localize';
 import { ScmType } from './ScmType';
 import { SiteClient } from './SiteClient';
+import { ISiteTreeRoot } from './tree/ISiteTreeRoot';
 import { nonNullProp } from './utils/nonNull';
 
-export async function editScmType(client: SiteClient, node: AzureTreeItem): Promise<string | undefined> {
+export async function editScmType(ti: AzureParentTreeItem<ISiteTreeRoot>, newScmType?: ScmType): Promise<string | undefined> {
```

Comment:
We'll need to edit the scmType of apps without user input (ie connecting/disconnecting to a GitHub repo) so this is necessary. Without a scmType, it will run as per usual.

Reply:
I know this is related to the deployments stuff, but I don't understand the exact scenario. Is there an issue for this feature that you can link me to?

Reply:
I only have this: https://github.com/Microsoft/vscode-azureappservice/issues/677
But Matt has more details in the OneNote (not sure if I should link that here)

Reply:
@fiveisprime can you put more of the Deployments feature in Github issues?